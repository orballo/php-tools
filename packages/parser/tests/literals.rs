// mod common;

// use common::Extractor;
// use insta::{assert_yaml_snapshot, glob, with_settings};
// use parser::{PHPParser, PHPRule};
// use pest::Parser;
// use std::fs;

// #[test]
// fn single_quoted_string_literal() {
//     glob!("fixtures/single_quoted_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals = Extractor::default()
//             .extract_nodes(pairs, PHPRule::SINGLE_QUOTED_STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             assert_yaml_snapshot!(literals);
//         })
//     });
// }

// #[test]
// fn double_quoted_string_literal() {
//     glob!("fixtures/double_quoted_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals = Extractor::default()
//             .extract_nodes(pairs, PHPRule::DOUBLE_QUOTED_STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             insta::assert_yaml_snapshot!(literals);
//         })
//     });
// }

// #[test]
// fn heredoc_string_literal() {
//     glob!("fixtures/heredoc_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals = Extractor::default()
//             .extract_nodes(pairs, PHPRule::HEREDOC_STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             insta::assert_yaml_snapshot!(literals);
//         })
//     });
// }

// #[test]
// fn nowdoc_string_literal() {
//     glob!("fixtures/nowdoc_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals =
//             Extractor::default().extract_nodes(pairs, PHPRule::STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             insta::assert_yaml_snapshot!(literals);
//         })
//     });
// }
