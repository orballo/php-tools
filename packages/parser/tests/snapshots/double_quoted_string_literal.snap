---
source: packages/parser/tests/literals.rs
expression: literals
input_file: packages/parser/tests/fixtures/double_quoted_string_literals.php
---
- name: Simple string
  content: "\"This is a double quoted string literal\""
  line: 4
  length: 40
- name: String with common escape sequences
  content: "\"\\\\ \\\" \\$ \\e \\f \\n \\r \\t \\v\""
  line: 6
  length: 28
- name: String with octal escape sequences
  content: "\"\\7 \\77 \\777\""
  line: 8
  length: 13
- name: String with hexadecimal escape sequences
  content: "\"\\x8 \\xFF\""
  line: 10
  length: 10
- name: String with unicode escape sequences
  content: "\"\\u{91223} \\u{00CCFF} \\u{1}\""
  line: 12
  length: 28
- name: String with literal special characters
  content: "\"\\$efnrtv\""
  line: 14
  length: 10
- name: String with final backslash
  content: "\"\\\\\""
  line: 16
  length: 4
- name: Empty string
  content: "\"\""
  line: 18
  length: 2
- name: String with variable
  content: "\"variable: $foo\""
  line: 20
  length: 16
